components/config/src/config/mod.rs

```diff
@@ -205,25 +205,36 @@ impl Config {
         }
     }
 
-    /// Adds the default language to the list of languages if not present
+    /// Adds the default language to the list of languages if options for it are specified at base level of config.toml.
+    /// If section for the same language also exists, the options at this section and base are merged and then adds it
+    /// to list.
     pub fn add_default_language(&mut self) {
         // We automatically insert a language option for the default language *if* it isn't present
         // TODO: what to do if there is like an empty dict for the lang? merge it or use the language
         // TODO: as source of truth?
-        if !self.languages.contains_key(&self.default_language) {
-            self.languages.insert(
-                self.default_language.clone(),
-                languages::LanguageOptions {
-                    title: self.title.clone(),
-                    description: self.description.clone(),
-                    generate_feed: self.generate_feed,
-                    feed_filename: self.feed_filename.clone(),
-                    build_search_index: self.build_search_index,
-                    taxonomies: self.taxonomies.clone(),
-                    search: self.search.clone(),
-                    translations: self.translations.clone(),
-                },
-            );
+
+        let mut base_default_language_options = languages::LanguageOptions {
+            title: self.title.clone(),
+            description: self.description.clone(),
+            generate_feed: self.generate_feed,
+            feed_filename: self.feed_filename.clone(),
+            build_search_index: self.build_search_index,
+            taxonomies: self.taxonomies.clone(),
+            search: self.search.clone(),
+            translations: self.translations.clone(),
+        };
+
+        if !base_default_language_options.is_default() {
```

Comment:
This logic is wonky i think? you're checking whether some values are set on the struct you just made up.

You should check grab the existing struct in `self.languages` if it exists and handle all potential conflicts in this function to keep it simple.
I don't know if i'm clear, let me know if it isn't and I can do a quick version of what I think

Reply:
I think I get what you mean. The issue with my logic is that it's a bit hard to follow, even if it would turn out to be sound. I'm checking whether the struct I just made up from the base language data is of default value in order to know if the user has specified any data at base, since this data always will be present.

We could get rid of this step, and just merge if data is specified at section. I'd have to change the warning message, because right now it warns the user if they've specified data for the default language at base _and_ under a section. The new warning would pop up every time the user has specified data under a section that is of the default language, since base data is always available as default. If specifying default language data under a section isn't frowned upon though, I can simply get rid of the warning and only raise an error if there's a merge conflict.

Reply:
Actually, you're right, the logic is incorrect. I'll fix it, and I think I also have a way of getting the behavior you desire in a quite simple manner.